examples/update-states.js

```diff
@@ -0,0 +1,103 @@
+const async = require('async');
+const Redis = require('ioredis');
+
+/*
+This script updates the state of Utapi `numberOfObjects` and `storageUtilized`
+starting from the current timestamp (i.e., the latest 15 minute interval).
+
+To use:
+* Set REDIS_ENDPOINT to the host and port where the Redis server is running
+* Set the various metric level states (BUCKET_STATES, ACCOUNT_STATES,
+  USER_STATES, SERVICE_STATE) to your desired values. This can be an array of n
+  length of objects. Each value `storageUtilized` and `numberOfObjects` of the
+  state objects is optional, and you should only include those that you want to
+  change. To skip updating a particular metric level state, the state should be
+  set to an empty array.
+*/
+
+const REDIS_ENDPOINT = {
+    host: '127.0.0.1',
+    port: 6379,
+};
+const BUCKET_STATES = [
+    {
+        bucket: '<bucket-name>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    }
+];
+const ACCOUNT_STATES = [
+    {
+        accountId: '<canonical-id>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    }
+];
+const USER_STATES = [
+    {
+        userId: '<user-id>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    }
+];
+const SERVICE_STATE = [
+    {
+        service: '<service-name>',
+        storageUtilized: 0,
+        numberOfObjects: 0,
+    }
+];
+
+const allStates = {
+    buckets: BUCKET_STATES,
+    accounts: ACCOUNT_STATES,
+    users: USER_STATES,
+    service: SERVICE_STATE,
+}
+
+const allParams = [];
+Object.keys(allStates).forEach(key => allStates[key].forEach(state =>
+    allParams.push(Object.assign(state, { service: 's3', level: key }))));
+
+function generateStateKey(params, metric) {
+    const { bucket, accountId, userId, level, service } = params;
+    const id = bucket || accountId || userId || service;
+    return `${service}:${level}:${id}:${metric}`;
+}
+
+function getCurrrentTimestamp() {
+    const time = new Date(Date.now());
+    const minutes = time.getMinutes();
+    return time.setMinutes((minutes - minutes % 15), 0, 0);
+}
+
+const redisClient = new Redis(Object.assign({
+    enableOfflineQueue: true,
+    keepAlive: 3000,
+}, REDIS_ENDPOINT));
+
+async.each(allParams, (params, cb) => {
+    const { storageUtilized, numberOfObjects } = params;
+    const timestamp = getCurrrentTimestamp();
+    const cmds = [];
+    if (storageUtilized !== undefined) {
+        const storageUtilizedKey = generateStateKey(params, 'storageUtilized');
+        cmds.push(
+            ['zremrangebyscore', storageUtilizedKey, timestamp, timestamp],
```

Comment:
not familiar with redis so just making sure here, did you want a 2d-array here?

Reply:
This is how ioredis module does a batch of commands. See https://github.com/luin/ioredis.

For example:

```js
redis.multi([
  ['set', 'foo', 'bar'],
  ['get', 'foo']
]).exec(function () { /* ... */ });
```